packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -230,7 +235,7 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     }
 
     final int lastIndex = indexOf(lastChild);
-    final double leadingScrollOffset = indexToLayoutOffset(itemExtent, firstIndex);
+    final double leadingScrollOffset = firstIndex == 0 ? 0 : indexToLayoutOffset(itemExtent, firstIndex - 1) + itemExtent;
```

Comment:
The `firstIndex == 0 ? 0` also feels strange. It seems to half-encode into this class that there may be a padding proceeding the first element of the list, but nothing else in this class is doing that and it is instead the task of a subclass to define any kind of padding.